### PR TITLE
[media-library]: include more error information in native rejections

### LINKED ISSUE #30441: expo-media-library is throwing an error without any useful details
### Minimal reproducible example

N/A

### What platform(s) does this occur on?

iOS

### Where did you reproduce the issue?

in a standalone app

### Summary

I am using expo-media-library in a production iOS app. Some of my users run into this error when deleting photos: "Couldn't remove assets"

Looking at the expo source code, I can see that this error message corresponds with this error: https://github.com/expo/expo/blob/0797cb28e98e5fe34f04e97fc9a24fc5f8740545/packages/expo-media-library/ios/MediaLibraryExceptions.swift#L69

Which is referenced here: https://github.com/expo/expo/blob/0797cb28e98e5fe34f04e97fc9a24fc5f8740545/packages/expo-media-library/ios/MediaLibraryModule.swift#L210

Is it possible to get some additional error info along with this message? As it currently stands, there is absolutely no information to go off of in order to debug the issue.

### Environment

N/A

### Expo Doctor Diagnostics

N/A

### PR DESCRIPTION
**Pull request by vonovak**
# Why

closes #30441

# How

the native module rejections include the error message from the native error

# Test Plan

#30441 mentions issues with deleting assets. Unfortunately I wasn't able to get the delete method to report an error (I tried deleting already deleted asset, or provide invalid data), but since it compiles and tests in bare expo pass I believe it's safe.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-media-library/ios/MediaLibraryUtilities.swift [hunk 1]
```diff
@@ -55,7 +55,7 @@ func stringifyAlbumType(type: PHAssetCollectionType) -> String {
 
 func exportAssetInfo(asset: PHAsset) -> [String: Any?]? {
   if var assetDict = exportAsset(asset: asset) {
-    assetDict["location"] = exportLocation(location: asset.location)
+    assetDict["location"] = exportLocation(location: asset.location) ?? NSNull()
```
Comment (vonovak) [inline packages/expo-media-library/ios/MediaLibraryUtilities.swift:58]: [this assertion was failing](https://github.com/expo/expo/blob/8b857d8e6741696ed74a783d86454d616d17396b/apps/test-suite/tests/MediaLibrary.js#L197) - the `location` entry wasn't present in the response. now it's `null`.

File: packages/expo-media-library/ios/MediaLibraryUtilities.swift [hunk 2]
```diff
@@ -218,7 +218,7 @@ func exportCollection(_ collection: PHAssetCollection?, folderName: String? = ni
     "assetCount": assetCountOfCollection(collection),
     "startTime": exportDate(collection.startDate),
     "endTime": exportDate(collection.endDate),
-    "approximateLocation": exportLocation(location: collection.approximateLocation),
+    "approximateLocation": exportLocation(location: collection.approximateLocation) ?? NSNull(),
```
Comment (alanjhughes) [inline packages/expo-media-library/ios/MediaLibraryUtilities.swift:221]:
```suggestion
    "approximateLocation": exportLocation(location: collection.approximateLocation) ?? nil,
```
Any reason we wouldn't use `nil` here?

**Review by alanjhughes — APPROVED** [review]

File: packages/expo-media-library/ios/MediaLibraryUtilities.swift (hunk 1 above)
Comment (behenate) [reply packages/expo-media-library/ios/MediaLibraryUtilities.swift:58]: I think it's a good idea, this is a typical use case for the `Record`

**Review by behenate — APPROVED** [review]

File: packages/expo-media-library/ios/MediaLibraryUtilities.swift (hunk 2 above)
Comment (behenate) [reply packages/expo-media-library/ios/MediaLibraryUtilities.swift:221]: I think we should, the code would be a lot more readable

**Pull request merged**